v2/yarpcthrift/register.go

```diff
@@ -76,15 +76,20 @@ func BuildProcedures(s Service, opts ...RegisterOption) []yarpc.EncodingProcedur
 	for _, method := range s.Methods {
 		var spec yarpc.EncodingHandlerSpec
 		spec = yarpc.NewUnaryEncodingHandlerSpec(unaryEncodingHandler{h: method.Handler})
-		codec := func() yarpc.InboundCodec {
-			return newCodec(proto, rc.Enveloping)
-		}
+
+		// In thrift the error codec and body codec need to be the same, as errors are
+		// serialized differently based on enveloping/protocol of the incoming requests.
+		codec := newCodec(proto, rc.Enveloping)
```

Comment:
This would actually fail our concurrency tests, because we need a codec per request, not per procedure. It might be difficult/maybe impossible if we want to share the responder between the codec and error codec.